Framework/Algorithms/src/AsymmetryCalc.cpp

```diff
@@ -129,6 +129,7 @@ void AsymmetryCalc::exec() {
   // Create a point data workspace with only one spectra for forward
   auto outputWS = DataObjects::create<API::HistoWorkspace>(
       *inputWS, 1, tmpWS->points(forward));
+  outputWS->getSpectrum(0).setDetectorID(static_cast<detid_t>(1));
```

Comment:
Isn't the modern way to do this via `setSpectrumDefinition` on `ExperimentInfo`?

Reply:
Well, I think that is exactly the problem: `setSpectrumDefinition` will not let you set anything that corresponds to detector IDs that do not exist.